### DISCUSSION TITLE
How to build a preview and update Slate state?

### DISCUSSION THREAD

**Discussion by bluebill1049** [ianstormtaylor/slate#4855] [Q&A]
Hi there,

according to the breaking change with version 0.54 https://docs.slatejs.org/general/changelog#0.54-december-12-2019, how can we build a controlled preview of what's user's typing? I think right now it's not possible to set state externally or async since the Slate component is uncontrolled. is there any API that we can update?

```tsx
const [value, setState] = React.useState([])

useEffect(() => {
  // setState aysnc/dynamicl won't work
}, [])

<Slate onChange={(value) => setState(value)} value={value} />
```

**Comment by francisashley** [comment]
~Anyone? This is such an obvious feature. How can we set the state remotely~?

Found this to work:

```
const [editor] = React.useState(() => withReact(createEditor()));
editor.children = value;
editor.onChange();
```
